portfolio-project-matching-app/backend/dao.js

```diff
@@ -271,6 +272,25 @@ const readAllDocs = async (coll) => {
     }
 }
 
+const readQuerySnapshotById = async (coll, field, id) => {
+    /*
+    DESCRIPTION:    retrieves collection snapshot from specified collection
+                    where field matches id
+
+    INPUT:          coll (string) : name of Firebase collection where the
+                    document being updated is located
+
+                    field (string): document field to be compared against id
+
+                    id (string) : document ID of document being updated
+
+    RETURN:         collection snapshot subject to criteria above
+    */
+    // get collection snapshot
+    const querySnap = await getCollectionSnapshotByCriteria(coll, field, '==', id);
+    return querySnap;
+}
+
```

Comment:
Might be a bit unnecessary upon reviewing things, but probably alright to leave in.